__tests__/should-close-popup-on-unmount.js

```diff
@@ -1,27 +1,34 @@
 import React from 'react';
-import renderer from 'react-test-renderer';
-import OauthPopup from "../src/index.js";
+import renderer, { act } from 'react-test-renderer';
+import OauthPopup from '../src/index';
 
 test('OauthPopup closes popup on unmount', () => {
-    const close = jest.fn()
-    global.open = jest.fn().mockReturnValue({
-        location: "http://www.google.com",
-        close,
-    });
-    const component = renderer.create(
+  let component;
+  const close = jest.fn();
+  global.open = jest.fn().mockReturnValue({
+    location: {
+      href: 'http://www.google.com',
+    },
+    close,
+  });
+  act(() => {
+    component = renderer.create(
         <OauthPopup>
             <div>coooooooool</div>
             <h2>siiiiiiiick</h2>
             <h2>leeegiiiiit</h2>
             <button>Doooooope</button>
-        </OauthPopup>
+        </OauthPopup>,
     );
-    let popup = component.toJSON();
-    expect(popup).toMatchSnapshot();
+  });
+  let popup = component.toJSON();
+  expect(popup).toMatchSnapshot();
+  act(() => {
     popup.props.onClick();
-    popup = component.toJSON();
-    expect(global.open.mock.calls.length).toBe(1);
-    component.unmount();
-    expect(close.mock.calls.length).toBe(1);
-    expect(global.open.mock.calls.length).toBe(1);
-});
\ No newline at end of file
+  });
+  popup = component.toJSON();
+  expect(global.open.mock.calls.length).toBe(1);
```

Comment:
[minor] You can also something like `expect(global.open).toHaveBeenCalled()` See: https://jestjs.io/docs/en/expect#tohavebeencalledtimesnumber